File: frontend/src/lib/backend/types/genrePublication.ts
export interface GetGenrePublicationResponse {
  Genres: GenrePublication[]
  Seasons: Season[]
}

export interface GenrePublication {
  GenreID: number
  GenreName: string
  DisplayOrder: number
  SeasonTeams: SeasonTeam[]
}

export interface SeasonTeam {
  ID: number
  Teams: TeamPublicationStatus[]
}

export interface TeamPublicationStatus {
  ID: number
  IsPublished: boolean
}

export interface Season {
  ID: number
  Number: number
  Teams: Team[]
}

export interface Team {
  ID: number
  Name: string
}

export interface UpdateGenrePublicationRequest {
  GenreID: number
  SeasonID: number
  TeamID: number
  IsPublished: boolean
}

File: backend/database/models/genre_publication.go
package models

import "gorm.io/gorm"

type GenrePublication struct {
	gorm.Model
	ID          uint   `gorm:"primaryKey"`
	GenreID     uint   `gorm:"not null;uniqueIndex:idx_genre_publication"`
	SeasonID    uint   `gorm:"not null;uniqueIndex:idx_genre_publication"`
	TeamID      uint   `gorm:"not null;uniqueIndex:idx_genre_publication"`
	Genre       Genre  `gorm:"foreignKey:GenreID"`
	Season      Season `gorm:"foreignKey:SeasonID"`
	Team        Team   `gorm:"foreignKey:TeamID"`
	IsPublished bool   `gorm:"not null"`
}
